org.syphon.Syphon.json

```diff
@@ -0,0 +1,98 @@
+{
+    "app-id": "org.syphon.Syphon",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "41",
+    "sdk": "org.gnome.Sdk",
+    "command": "syphon",
+    "separate-locales": false,
+    "finish-args": [
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--socket=pulseaudio",
+        "--socket=session-bus",
```

Comment:
According to https://github.com/flathub/flathub/wiki/App-Requirements#dbus-access, I shouldn't be using this. However, without this the app dies with:

```
Failed to register: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: org.freedesktop.DBus.Error.ServiceUnknown
```
According to https://github.com/elementary/docs/issues/120, this is the solution.

Please advise.

Reply:
run with `--log-session-bus` (this is super verbose) and look at the address' that are actually called. allowlist the required ones.

Reply:
I've figured out that the `Failed to register` error was down to the binary having a different package name (old name) to the flatpak. Patch submitted upstream: https://github.com/syphon-org/syphon/pull/450. However, I'm still having issues with some features breaking without `--socket=system-bus`. I believe I found a possible cause, this being the [`connectivity_plus`](https://github.com/fluttercommunity/plus_plugins/blob/main/packages/connectivity_plus/connectivity_plus_linux/lib/src/connectivity.dart) dep which depends on [`nm.dart`](https://github.com/canonical/nm.dart/blob/442cac055ac44d5610e55f45e8360eb6e57c6ac7/lib/src/network_manager_client.dart) which, in turn, attempts to use dbus to communicate with NetworkManager. I haven't been able to track down exactly what permissions it needs though. Would you mind taking a look?

Likely worth noting is the following error from the app logs:
```
[ERROR:flutter/lib/ui/ui_dart_state.cc(209)] Unhandled Exception: SocketException: Connection failed (OS Error: No such file or directory, errno = 2), address = /var/run/dbus/system_bus_socket, port = 0
```

Reply:
Like the session bus you can log the system bus with `--log-system-bus --socket=system-bus` to find which `--system-talk-name=`

Reply:
Aha, I was using `--talk-name` instead of `--system-talk-name`. That solved it. Thanks!